Types/Table.php

```diff
@@ -62,4 +65,52 @@ public static function alias(array $tokens) {
         if ($operation === SqlOperations::UPDATE) return $tokens['UPDATE'][0]['alias']['name'];
         return $tokens['FROM'][0]['alias']['name'];
     }
+
+    /**
+     * replaces the table in the tokens array with the given table
+     *
+     * @param  array $tokens
+     * @param  array $newTable
+     * @return array
+     */
+    public static function replace(array $tokens, $newTable) {
+        Assertions::assertHashMap('tokens', $tokens);
+
+        $operation = SqlOperation::create($tokens);
+        if ($operation === SqlOperations::INSERT) return self::replaceTableName($tokens, $newTable, 'INSERT', 1);
+        return $operation === SqlOperations::UPDATE ? self::replaceTableName($tokens, $newTable, 'UPDATE', 0) : self::replaceTableName($tokens, $newTable, 'FROM', 0);
```

Comment:
wofür is das mit dem first und secondkey als argument? soweit ich das sehe, is aus "INSERT" etc ableitbar, welcher key (oder wird das noch anders verwendet)?